src/Nethermind/Nethermind.Blockchain/TransactionPools/TransactionPool.cs

```diff
@@ -0,0 +1,218 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Logging;
+using Nethermind.Core.Model;
+using Nethermind.Dirichlet.Numerics;
+
+namespace Nethermind.Blockchain.TransactionPools
+{
+    public class TransactionPool : ITransactionPool
+    {
+        private static readonly Random GlobalRandom = new Random();
+        [ThreadStatic] private static Random _random;
+
+        private readonly ConcurrentDictionary<Keccak, Transaction> _pendingTransactions =
+            new ConcurrentDictionary<Keccak, Transaction>();
+
+        private readonly ConcurrentDictionary<Type, ITransactionFilter> _filters =
+            new ConcurrentDictionary<Type, ITransactionFilter>();
+
+        private readonly ITransactionStorage _transactionStorage;
+        private readonly IReceiptStorage _receiptStorage;
+
+        private readonly ConcurrentDictionary<PublicKey, ISynchronizationPeer> _peers =
+            new ConcurrentDictionary<PublicKey, ISynchronizationPeer>();
+
+        private readonly IEthereumSigner _signer;
+        private readonly ILogger _logger;
+        private readonly int _peerThreshold;
+
+        public TransactionPool(ITransactionStorage transactionStorage, IReceiptStorage receiptStorage,
+            IEthereumSigner signer, ILogManager logManager, int peerThreshold = 20)
```

Comment:
peer threshold - I would use the same as in sync manager - we could share NetworkConfig here

Reply:
Ok